### PR TITLE
Make the prow/heart plugin configurable

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Repositories where the `kubernetes-merge-robot` is not active would also
like to share the love.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @spxtr 
/area prow

### CODE REVIEW

File: prow/plugins/heart/heart.go [hunk 1]
```diff
@@ -23,13 +23,13 @@ import (
...
 	"github.com/Sirupsen/logrus"
 
+	"k8s.io/test-infra/prow/config"
 	"k8s.io/test-infra/prow/github"
 	"k8s.io/test-infra/prow/plugins"
 )
 
 const (
 	pluginName     = "heart"
-	botName        = "k8s-merge-robot"
```
Comment (spxtr) [inline prow/plugins/heart/heart.go]: Please add this as an adoree in `prow/config.yaml`.

Comment (stevekuznetsov) [inline prow/plugins/heart/heart.go]: Ok!

File: prow/plugins/heart/heart.go [hunk 2]
```diff
@@ -53,37 +53,61 @@ type githubClient interface {
...
 
-func handleIC(gc githubClient, log *logrus.Entry, ic github.IssueCommentEvent) error {
+func handleIC(c client, ic github.IssueCommentEvent) error {
 	// Only consider new comments on PRs.
 	if !ic.Issue.IsPullRequest() || ic.Action != "created" {
 		return nil
 	}
-	// Only consider the merge bot.
-	if ic.Comment.User.Login != botName {
+	validLogin := false
```
Comment (spxtr) [inline prow/plugins/heart/heart.go]: It's not so much that it's a valid login, but instead it's an adored login.

**Comment by spxtr** [comment]
Looks good. Please bump hook.

**Comment by stevekuznetsov** [comment]
@spxtr updated `validLogin` --> `adoredLogin` and bumped `hook`

**Comment by stevekuznetsov** [comment]
@spxtr how did you determine that I needed to bump `hook`? Is it possible to write a validation that would automatically let me know I changed code but did not do the appropriate bump? If I could write that it might save you a little time :)

**Comment by spxtr** [comment]
All plugin changes need a hook bump. Eventually we'll have bazel decide when we need to make changes.
```shell
$ bazel query "kind(.*_binary, rdeps(//..., set(//prow/plugins/heart:go_default_library)))"
//prow/cmd/hook:hook
```

File: prow/config.yaml [hunk 3]
```diff
@@ -18,6 +18,10 @@ triggers:
   - google/cadvisor
   trusted_org: kubernetes
 
+heart:
```
Comment (kargakis) [inline prow/config.yaml]: Thinkins out loud, wouldn't it make sense to extend prow/plugins.yaml for plugin-specific config instead of this?

Comment (spxtr) [inline prow/config.yaml]: Yeah, something like that makes sense. I'd like to redo much of prow's config, but I wanted to get a few examples of plugin configs first. Now is probably a good time to revisit that idea.

**Pull request merged**